app/scripts/app-init.js

```diff
@@ -184,11 +184,22 @@ const registerInPageContentScript = async () => {
   }
 };
 
+// On MV3 builds we must listen for this event in `app-init`, otherwise we found that the listener
+// is never called.
+// For MV2 builds, the listener is added in `background.js` instead.
 chrome.runtime.onInstalled.addListener(function (details) {
   if (details.reason === 'install') {
-    chrome.storage.session.set({ isFirstTimeInstall: true });
-  } else if (details.reason === 'update') {
-    chrome.storage.session.set({ isFirstTimeInstall: false });
+    // This condition is for when `background.js` was loaded before the `onInstalled` listener was
```

Comment:
since we don't handle `update` case now, is it worth uninstalling these listeners after we detect an `install`?

Reply:
Good idea

Reply:
Done here: 01c3b34430